src/gui/screen_lan_settings.c

```diff
@@ -80,12 +80,12 @@ static void _addrs_to_str(char *param_str, uint8_t flg) {
             ip4_addr_str, ip4_msk_str, ip4_gw_str, param_str);
 }
 
-static void _parse_MAC_addr(char *mac_addr_str) {
+static void _parse_MAC_addr(char *mac_addr_str, uint32_t mac_addr_str_size) {
     volatile uint8_t mac_addr[] = { 0, 0, 0, 0, 0, 0 };
     for (uint8_t i = 0; i < MAC_ADDR_SIZE; i++)
         mac_addr[i] = *(volatile uint8_t *)(MAC_ADDR_START + i);
 
-    sprintf(mac_addr_str, "%x:%x:%x:%x:%x:%x", mac_addr[0], mac_addr[1], mac_addr[2], mac_addr[3], mac_addr[4], mac_addr[5]);
+    snprintf(mac_addr_str, mac_addr_str_size, "%x:%x:%x:%x:%x:%x", mac_addr[0], mac_addr[1], mac_addr[2], mac_addr[3], mac_addr[4], mac_addr[5]);
```

Comment:
We might consider using assertions or other ways of making sure we „fail safely“ if the mac_add_str_size is not sufficient.

Reply:
_Static_assert - gcc specific C-style compile-time assert, probably non-portable

Reply:
yeah, I think we have to go with a runtime check